src/cluster.c

```diff
@@ -1025,6 +1024,13 @@ clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, in
         margc = ms->commands[i].argc;
         margv = ms->commands[i].argv;
 
+        /* Only valid for sharded pubsub as regular pubsub can operate on any node and bypasses this layer. */
+        if (!pubsubshard_included &&
```

Comment:
@cjx-zar please note these code.
we fake a multi for non-multi.
```
        /* In order to have a single codepath create a fake Multi State
         * structure if the client is not in MULTI/EXEC state, this way
         * we have a single codepath below. */
        ms = &_ms;
        _ms.commands = &mc;
        _ms.count = 1;
        mc.argv = argv;
        mc.argc = argc;
        mc.cmd = cmd;
```

Reply:
What I mean is that even if you fake a multi, the flag of 'sunsubscribeCommand' is CMD_CHANNEL_UNSUBSCRIBE, which will not make pubsubshard_included become 1, so it will return MOVED when executed on the replica, which is inconsistent with the behavior of the previous version.
```
ChannelSpecs commands_with_channels[] = {
    {sunsubscribeCommand, CMD_CHANNEL_UNSUBSCRIBE, 1, -1},
    {NULL,0}
};
```

Reply:
@cjx-zar thanks, feel like you're right, can you make a PR to fix it and make a test to cover it?

Reply:
Sure :)